core/src/main/scala/quasar/destination/gbq/GBQDestinationModule.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.gbq
+
+import argonaut._, Argonaut._
+
+import cats.effect.{
+  Concurrent, 
+  ConcurrentEffect, 
+  ContextShift, 
+  Resource, 
+  Timer
+}
+import cats.data.EitherT
+import cats.implicits._
+
+import org.http4s.{
+  AuthScheme,
+  Credentials,
+  Method,
+  Request,
+  Status,
+  Uri}
+import org.http4s.headers.Authorization
+import org.http4s.client.Client
+import org.slf4s.Logging
+
+import quasar.api.destination.{DestinationType, DestinationError}
+import quasar.api.destination.DestinationError.InitializationError
+import quasar.connector.destination.{Destination, DestinationModule}
+import quasar.connector.MonadResourceErr
+
+import scala.Predef._
+import scala.{
+  StringContext,
+  Some,
+  Either,
+  Unit
+}
+import scala.concurrent.ExecutionContext
+
+object GBQDestinationModule extends DestinationModule with Logging {
+
+  val Redacted: String = "<REDACTED>"
+
+  def destinationType = DestinationType("gbq", 1L)
+
+  def sanitizeDestinationConfig(config: Json) = {
+    config.as[GBQConfig].toOption match {
+      //TODO: maybe redact the entire service account key
```

Comment:
Do we need to make a decision about this now?

Reply:
In my initial look through a service account key json file the `private_key` seemed like the most important thing to redact. Looking at it again maybe these values should also be redacted: 

`type`
`private_key_id`
`project_id`
`client_email` <-- contains project id
`client_id`
`client_x509_cert_url` <-- contains project id

If we redact these there's like 3 or 4 things left, so maybe just redact the whole thing?

Reply:
sample config:
```
{
  "type": "service_account",
  "project_id": "id123",
  "private_key_id": "sdakfjae",
  "private_key": "-----BEGIN PRIVATE KEY----- ... -----END PRIVATE KEY-----\n",
  "client_email": "bigquer-service-account@place.internet",
  "client_id": "28374239874",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/account-thingz"
}
```

Reply:
@nicflores is what I pasted above the contents of `"authCfg": <service-account-json-auth-file-contents>`?

Looking at the destination config:
```
  "config": {
    "authCfg": <service-account-json-auth-file-contents>,
    "project": <existing-google-cloud-project>,
    "datasetId": <dataset-name>
  }
```
which of these values will be provided directly by the user (i.e. the user enters these values in the Precog UI when configuring the destination), and which will be gathered by other means (e.g. as constants hardcoded in the destination)? Does the user enter individual components of `<service-account-json-auth-file-contents>` in the Precog UI?

Reply:
@alissapajer, the README needs and update. I've now updated it. The config now looks like:
```
  "config": {
    "authCfg": <service-account-json-auth-file-contents>,
    "datasetId": <dataset-name>
  }
```
The only two things a user would need to pass in is the entire content of their service account file and a name for the dataset. The user doesn't need enter individual parts of the service account file, they can just paste the entire json. The project won't be needed since the project is embedded in the service account json file.

Reply:
How does the user get their service account file? Is this something they can easily get, in its entirety, from the BigQuery website?

Assuming they get this thing in its entirety and then just paste it in, let's consider the whole of `<service-account-json-auth-file-contents>` sensitive, thus redacting all of it, since this whole blob represents the authentication.

Reply:
Yeah its pretty straightforward, you get it from BigQuery and we'll include instructions on how to get this in the form in the UI.

Reply:
Definitely sensitive.

Reply:
Basically go here https://console.cloud.google.com/iam-admin/serviceaccounts, click your project and click "Create service account"